eng/automation/changelog/src/main/java/com/azure/resourcemanager/tools/changelog/changelog/DefinitionStageChangeLog.java

```diff
@@ -76,34 +79,22 @@ protected void calcChangeLog() {
                 if (newMethodStages.get(i).contains(method)) {
                     switch (method.getChangeStatus()) {
                         case REMOVED:
-                            addClassTitle(breakingChange);
-                            breakingChange.add(String.format("* `%s` was removed in stage %d", MethodName.name(method.getOldMethod().get()), i + 1));
+                            breakingChange.addMethodLevelChange(String.format("`%s` was removed in stage %d", MethodName.name(method.getOldMethod().get()), i + 1));
                             break;
                         case MODIFIED:
                             if (!method.getOldMethod().get().getLongName().equals(method.getNewMethod().get().getLongName())) {
-                                addClassTitle(breakingChange);
-                                breakingChange.add(String.format("* `%s` -> `%s` in stage %d", MethodName.name(method.getOldMethod().get()), MethodName.name(method.getNewMethod().get()), i + 1));
+                                breakingChange.addMethodLevelChange(String.format("`%s` -> `%s` in stage %d", MethodName.name(method.getOldMethod().get()), MethodName.name(method.getNewMethod().get()), i + 1));
                             }
                             break;
                     }
                 } else if (method.getOldMethod().isPresent()) {
-                    addClassTitle(breakingChange);
-                    breakingChange.add(String.format("* `%s` was removed in stage %d", MethodName.name(method.getOldMethod().get()), i + 1));
+                    breakingChange.addMethodLevelChange(String.format("`%s` was removed in stage %d", MethodName.name(method.getOldMethod().get()), i + 1));
                 }
             }
         }
         if (newSize > oldSize) {
             List<String> newStages = IntStream.range(oldSize + 1, newSize + 1).boxed().map(Object::toString).collect(Collectors.toList());
-            addClassTitle(breakingChange);
-            breakingChange.add(String.format("* Stage %s was added", String.join(", ", newStages)));
-        }
-    }
-
-    @Override
-    protected void addClassTitle(List<String> list) {
-        if (list.isEmpty()) {
-            list.add(String.format("#### `%s` was modified", parentClass));
-            list.add("");
+            breakingChange.addMethodLevelChange(String.format("Stage %s was added", String.join(", ", newStages)));
```

Comment:
nit, should we say `Required stage %s was added` (as only this kind of "add" breaks)

Reply:
changed